File: baekjoon/bronze/level1/test_1259/Main.java
package bronze.level1.test_1259;

import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String input = br.readLine();

        while (!input.equals("0")) {
            String answer = "yes";
            for (int i = 0; i < input.length() / 2; i++) {
                if (input.charAt(i) != input.charAt(input.length() - i - 1)) {
                    answer = "no";
                    break;
                }
            }
            input = br.readLine();
            bw.write(answer);
            bw.newLine();
        }
        bw.close();
    }
}

File: baekjoon/bronze/level1/test_1259/kotlin/Main.kt
package baekjoon.bronze.level1.test_1259.kotlin

fun main() {
    var input = readln()

    while (input != "0") {
        val arr = input.toCharArray()

        var answer = "yes"
        for (i in 0 until arr.size / 2) {
            if (arr[i] != arr[arr.size - i - 1]) {
                answer = "no"
                break
            }
        }

        println(answer)
        input = readln()
    }
}
